### DISCUSSION TITLE
API is not working only on mobile

### DISCUSSION THREAD

**Discussion by xup6128** [emailjs-com/help#39] [Q&A] [answered]
Hi I am using emailJS to send email on my web, everything is fine on desktop. 

But If I using my mobile to open the same web, then API will failed and return status '0' with empty text.

I also activate 'Allow EmailJS API for non-browser applications' of my account.

Would anyone please give me a hand.

I google which may be a CORS problem, but I don't see any setting on dashboard.

here is my web: https://xup6128.github.io/web_camera/

GitHub: https://github.com/xup6128/web_camera

Thanks for all

**Comment by xr0master** [comment]
It sounds like you haven't overridden the default form action by using the preventDefault method.

**Reply by xup6128** [reply]
I'm not sending with form but params only, not sure that where I should add preventDefault


```html
     <div id="container">
        <video autoplay="true" id="videoElement" playsinline>
        </video>
        <button id="downloadButton"></button>
        <div id="errorDisplay">console</div>
     </div>
```

``` js
  
      document.getElementById("downloadButton").addEventListener("click", function () {
        const canvas = createCanvas();
        sendEmail(canvas);
      });
      
    function sendEmail(canvas) {
        const templateParams = {
            to_email: userEmail,
            from_name: 'LIGHT',
            subject: 'Image',
            html: `<img src="${canvas.toDataURL('image/jpeg', 0.1)}" alt="Embedded Image`,
        };
    
    
        emailjs.send("service_xr70ero", "template_rzhk0bq", templateParams, "SXpe5bXZFtpUlLK6W")
            .then(function (response) {
                console.log("Success！", response.status, response.text);
            }, function (error) {
                console.log("Failed...", error);
            });
    }
```

**Reply by xr0master** [reply, marked as answer]
I see. The status '0' means the connection was terminated. This usually means that the page has reloaded. Do you have any idea what some code in the app could do this?

**Reply by xup6128** [reply]
Oh yes, cause I'm also download image with link click. Once I removed it, email send successfully.

Thanks for your time and feedback,
wish you have a good day
